### PR TITLE
[java] Use @JsonFormat instead of @JsonSerialize for bigDecimalAsString property

### PR DESCRIPTION
**Pull request by andeb**
Use `@JsonFormat` instead of `@JsonSerialize` for `bigDecimalAsString` so that `JsonGenerator.Feature.WRITE_BIGDECIMAL_AS_PLAIN` is respected and generates a `String` using `BigDecimal#toPlainString` instead of `BigDecimal#toString` if the feature is enabled.

For instance, consider the OpenAPI with a field definition as:

```
      BigDecimal:
        type: string
        format: number
```

The code generated will have the annotation `@JsonFormat(shape = JsonFormat.Shape.STRING)`, hence the client writing a POJO as string will see the value of the `BigDecimal` as plain string if the feature is enabled on Jackson.

```
FormatTest formatTest = new FormatTest();
formatTest.setBigDecimal(BigDecimal.valueOf(0,10));

ObjectMapper objectMapper = new ObjectMapper();
objectMapper.enable(JsonGenerator.Feature.WRITE_BIGDECIMAL_AS_PLAIN);

// will output "0.0000000000" as WRITE_BIGDECIMAL_AS_PLAIN is enabled, otherwise it outputs "0E-10"
System.out.println(objectMapper.writeValueAsString(formatTest));
```

### PR checklist
 
- [X] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [X] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [X] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh ./bin/configs/*.yaml
  ./bin/utils/export_docs_generators.sh
  ``` 
  (For Windows users, please run the script in [Git BASH](https://gitforwindows.org/))
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  IMPORTANT: Do **NOT** purge/delete any folders/files (e.g. tests) when regenerating the samples as manually written tests may be removed.
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (upcoming 7.6.0 minor release - breaking changes with fallbacks), `8.0.x` (breaking changes without fallbacks)
- [X] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

Mentions: @cachescrubber @welshm @MelleD @atextor @manedev79 @javisst @borsch @banlevente @Zomzog @martin-mfg @wing328

### CODE REVIEW

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractJavaCodegen.java [hunk 1]
```diff
@@ -568,8 +568,7 @@ public void processOpts() {
         this.sanitizeConfig();
 
         // optional jackson mappings for BigDecimal support
-        importMapping.put("ToStringSerializer", "com.fasterxml.jackson.databind.ser.std.ToStringSerializer");
-        importMapping.put("JsonSerialize", "com.fasterxml.jackson.databind.annotation.JsonSerialize");
+        importMapping.put("JsonFormat", "com.fasterxml.jackson.annotation.JsonFormat");
```
Comment (wing328) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractJavaCodegen.java:571]: how will this change impact the auto-generated code if bigDecimalAsString is not enabled (set to false by default)?

shall we add a if-then-else to check if bigDecimalAsString is enabled and update the importMapping accordingly?

Comment (andeb) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractJavaCodegen.java:571]: It's a good point. I thought `importMapping` was conditional to the import being added by `model.imports.add("JsonFormat");` (https://github.com/OpenAPITools/openapi-generator/pull/19322/files#diff-a98cba8ecadaf5a662edc508aa5857972339552ec6bd6e4216680c7c6070e643R1710)

I can also see that there some ad-hoc places where it manually removes this import from other mappings, like https://github.com/OpenAPITools/openapi-generator/blob/d1b9f9284a007da3a6f92fcad5fc966709be5454/modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/JavaCXFServerCodegen.java#L230-L231), so it seems a good change to add an if at this point.

I am not convinced that `JsonSerialize` is used only for BigDecimal support as the comment seems to indicate. Will have a look at this as well.
Thanks for your comment @wing328

Comment (andeb) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractJavaCodegen.java]: @wing328 I have pushed some commits that I believe resolve your concerns, let me know your thoughts

**Comment by martin-mfg** [comment]
Hi @andeb,

Searching the whole project for "ToStringSerializer" still yields some results. e.g.
- https://github.com/OpenAPITools/openapi-generator/blob/8f1d59fb0a0d6211e8d58555e769c316be70b556/modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/JavaHelidonClientCodegen.java#L375
- https://github.com/OpenAPITools/openapi-generator/blob/49d5cfc44104dc703663bee3c443c7ff1b17ced1/modules/openapi-generator/src/test/java/org/openapitools/codegen/java/JavaClientCodegenTest.java#L2657
- and others

It seems to me all of them can be removed, except for the matches in the `doesNotGenerateJacksonToStringSerializerAnnotation_whenLibraryIsGson_andSerializeBigDecimalAsStringIsTrue` unit test.
Or am I wrong?

And some "JsonSerialize" occurrences also seem to be unnecessary now.

**Comment by andeb** [comment]
Hi @martin-mfg, thanks for the review! Very spot on. Looks like I missed few references.

Pushed a new commit removing references to `JsonSerialize` and `ToStringSerializer`, and refactored the test to reflect the changes proposed.

Would you be able to have another look? Let me know if there's something that you would like me to do.

**Review by martin-mfg — APPROVED** [review]

**Comment by wing328** [comment]
@andeb thanks for the PR, which has been merged

**Pull request merged**